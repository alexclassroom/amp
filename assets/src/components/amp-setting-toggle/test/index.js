/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { create } from 'react-test-renderer';

/**
 * Internal dependencies
 */
import { AMPSettingToggle } from '../';

describe('AMPSettingToggle', () => {
	it('matches snapshots', () => {
		let wrapper = create(
			<AMPSettingToggle
				checked={true}
				onChange={() => null}
				text={'My text'}
				title={'My title'}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();

		wrapper = create(
			<AMPSettingToggle
				checked={false}
				onChange={() => null}
				title={'My title'}
			/>
		);
		expect(wrapper.toJSON()).toMatchSnapshot();
	});

	it('has correct elements and text', () => {
		let { container } = render(
			<AMPSettingToggle
				title="My title"
				onChange={() => null}
				checked={false}
			>
				{'children'}
			</AMPSettingToggle>
		);

		expect(container.querySelector('h3').textContent).toBe('My title');
		expect(container.querySelector('p')).toBeNull();
		expect(container.querySelector('input:checked')).toBeNull();

		container = render(
			<AMPSettingToggle
				title="My title"
				onChange={() => null}
				checked={true}
				text="My text"
			>
				{'children'}
			</AMPSettingToggle>
		).container;

		expect(container.querySelector('h3').textContent).toBe('My title');
		expect(container.querySelector('p').textContent).toBe('My text');
		expect(container.querySelector('input:checked')).not.toBeNull();
	});

	it('renders title if it is a valid element', () => {
		const { container } = render(
			<AMPSettingToggle
				title={<h6>{'My title'}</h6>}
				onChange={() => null}
				checked={false}
			/>
		);

		expect(container.querySelector('h6').textContent).toBe('My title');
	});

	it('does not render title if nothing is passed', () => {
		const { container } = render(
			<AMPSettingToggle onChange={() => null} checked={false} />
		);

		expect(container.querySelector('h3')).toBeNull();
	});
});
