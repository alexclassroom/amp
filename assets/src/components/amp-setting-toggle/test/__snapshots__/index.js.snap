// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AMPSettingToggle matches snapshots 1`] = `
<div
  className="amp-setting-toggle"
>
  <div
    className="components-base-control components-toggle-control css-1wgfkcs-Wrapper-boxSizingReset-classes ej5x27r4"
  >
    <div
      className="components-base-control__field css-yj91it-StyledField ej5x27r3"
    >
      <div
        className="components-flex components-h-stack css-79etx6-View-Flex-base-ItemsRow e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="HStack"
      >
        <span
          className="components-form-toggle is-checked"
        >
          <input
            checked={true}
            className="components-form-toggle__input"
            id="inspector-toggle-control-0"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="components-form-toggle__track"
          />
          <span
            className="components-form-toggle__thumb"
          />
        </span>
        <label
          className="components-toggle-control__label"
          htmlFor="inspector-toggle-control-0"
        >
          <div
            className="amp-setting-toggle__label-text"
          >
            <h3>
              My title
            </h3>
            <p>
              My text
            </p>
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;

exports[`AMPSettingToggle matches snapshots 2`] = `
<div
  className="amp-setting-toggle"
>
  <div
    className="components-base-control components-toggle-control css-1wgfkcs-Wrapper-boxSizingReset-classes ej5x27r4"
  >
    <div
      className="components-base-control__field css-yj91it-StyledField ej5x27r3"
    >
      <div
        className="components-flex components-h-stack css-79etx6-View-Flex-base-ItemsRow e19lxcc00"
        data-wp-c16t={true}
        data-wp-component="HStack"
      >
        <span
          className="components-form-toggle"
        >
          <input
            checked={false}
            className="components-form-toggle__input"
            id="inspector-toggle-control-1"
            onChange={[Function]}
            type="checkbox"
          />
          <span
            className="components-form-toggle__track"
          />
          <span
            className="components-form-toggle__thumb"
          />
        </span>
        <label
          className="components-toggle-control__label"
          htmlFor="inspector-toggle-control-1"
        >
          <div
            className="amp-setting-toggle__label-text"
          >
            <h3>
              My title
            </h3>
          </div>
        </label>
      </div>
    </div>
  </div>
</div>
`;
